grid2op/tests/test_shedding.py

```diff
@@ -0,0 +1,95 @@
+
+    def tearDown(self) -> None:
+        self.env.close()
+
+    def test_shedding_parameter_is_true(self):
+        assert hasattr(self.env, "allow_shedding")
+        assert self.env.allow_shedding is True
+
+    def test_shed_single_load(self):
+        # Check that a single load can be shed
+        load_idx = self.load_lookup["load_4_2"]
+        load_pos = self.env.load_pos_topo_vect[load_idx]
+        act = self.env.action_space({
+            "set_bus": [(load_pos, -1)]
+        })
+        obs, _, done, info = self.env.step(act)
+        assert not done
+        assert info["is_illegal"] is False
+        assert obs.topo_vect[load_pos] == -1
+
+    def test_shed_single_generator(self):
+        # Check that a single generator can be shed
+        gen_idx = self.gen_lookup["gen_0_0"]
+        gen_pos = self.env.gen_pos_topo_vect[gen_idx]
+        act = self.env.action_space({
+            "set_bus": [(gen_pos, -1)]
+        })
+        obs, _, done, info = self.env.step(act)
+        assert not done
+        assert info["is_illegal"] is False
+        assert obs.topo_vect[gen_pos] == -1
+
+    def test_shed_multiple_loads(self):
+        # Check that multiple loads can be shed at the same time
+        load_idx1 = self.load_lookup["load_4_2"]
+        load_idx2 = self.load_lookup["load_3_1"]
+        load_pos1 = self.env.load_pos_topo_vect[load_idx1]
+        load_pos2 = self.env.load_pos_topo_vect[load_idx2]
+        act = self.env.action_space({
+            "set_bus": [(load_pos1, -1), (load_pos2, -1)]
+        })
+        obs, _, done, info = self.env.step(act)
+        assert not done
+        assert info["is_illegal"] is False
+        assert obs.topo_vect[load_pos1] == -1
+        assert obs.topo_vect[load_pos2] == -1
+
+    def test_shed_load_and_generator(self):
+        # Check that load and generator can be shed at the same time
+                # Check that multiple loads can be shed at the same time
+        load_idx = self.load_lookup["load_4_2"]
+        gen_idx = self.gen_lookup["gen_0_0"]
+        load_pos = self.env.load_pos_topo_vect[load_idx]
+        gen_pos = self.env.gen_pos_topo_vect[gen_idx]
+        act = self.env.action_space({
+            "set_bus": [(load_pos, -1), (gen_pos, -1)]
+        })
+        obs, _, done, info = self.env.step(act)
+        assert not done
+        assert info["is_illegal"] is False
+        assert obs.topo_vect[load_pos] == -1
+        assert obs.topo_vect[gen_pos] == -1
+
+    def test_shedding_persistance(self):
+        # Check that components remains disconnected if shed
+        load_idx = self.load_lookup["load_4_2"]
+        load_pos = self.env.load_pos_topo_vect[load_idx]
+        act = self.env.action_space({
+            "set_bus": [(load_pos, -1)]
+        })
+        _ = self.env.step(act)
+        obs, _, done, _ = self.env.step(self.env.action_space({}))
+        assert not done
+        assert obs.topo_vect[load_pos] == -1
```

Comment:
you need also to test:
- Runner
- environment copied
- MultiMix environment
- TimedOutEnvironment
- MaskedEnvironment

(yes at this stage grid2op is far from "efficient"...)